src/Repository/TransactionRepository.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace AppBundle\Repository;
+
+use AppBundle\Entity\Transaction;
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Doctrine\Common\Persistence\ManagerRegistry;
+
+class TransactionRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, Transaction::class);
+    }
+
+    public function findByPayboxTransactionId(string $transactionId): ?Transaction
+    {
+        return $this->createQueryBuilder('t')
+            ->where('t.payboxTransactionId = :transactionId')
+            ->setParameters(['transactionId' => $transactionId])
+            ->getQuery()
+            ->getOneOrNullResult()
+        ;
+    }
+
+    /**
+     * @param string $emailAddress
+     *
+     * @return Transaction[]
+     */
+    public function findAllSuccessTransactionByEmail(string $emailAddress): array
```

Comment:
~~`SuccessTransaction`~~ -> `SuccessfulTransaction` 
because we say a successful transaction not a success transaction.